journal/journal.go

```diff
@@ -0,0 +1,120 @@
+package journal
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/filecoin-project/go-filecoin/clock"
+)
+
+// Writer defines an interface for recording events and their metadata
+type Writer interface {
+	// Write records an operation and its metadata to a Journal accepting variadic key-value
+	// pairs.
+	Write(event string, kvs ...interface{})
+}
+
+// Journal defines an interface for creating Journals with a topic.
+type Journal interface {
+	// Topic returns a Writer that records events for a topic.
+	Topic(topic string) Writer
+}
+
+// NewInMemoryJournal returns a journal backed by an in-memory map.
+func NewInMemoryJournal(clk clock.Clock) Journal {
```

Comment:
Thinking it might make more sense for this to go in a testing package as that's all I image we'll use it for, thoughts?

Reply:
I would also be open to a `NoopJournal`, getting this to have the same behaviour as the `ZapJournal` isn't simple.